### PR TITLE
Adds support for price rules

### PR DESCRIPTION
**Pull request by jordanliddle**
Adding support for price rules and discount codes

### CODE REVIEW

**Comment by justinstern** [comment]
This looks interesting. Are there any public docs yet on what this refers to?

**Comment by felix-d** [comment]
@jordanliddle I'll take a look tomorrow!

**Comment by kingscott** [comment]
Is this still an active WIP?

**Comment by jordanliddle** [comment]
Hey @kingscott it is - I'll be updating this branch early this week

File: test/price_rule_test.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    assert_equal -10, price_rule.value
+  end
+
+  def test_update_price_rule
+    price_rule_response = ActiveSupport::JSON.decode(load_fixture('price_rule'))
+    price_rule_response['price_rule']['value'] = -50.0
+    @price_rule.value = -50.0
+    fake 'price_rules/102586120', method: :put, status: 200, body: ActiveSupport::JSON.encode(price_rule_response)
+
+    assert_equal price_rule_response['price_rule']['value'], @price_rule.value
```
Comment (felix-d) [inline test/price_rule_test.rb]: I think you're missing `@price_rule.save`

File: test/discount_code_test.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    discount_code.save
+
+    assert_equal '{"discount_code":{"code":"SUMMERSALE10"}}', FakeWeb.last_request.body
+  end
+
+  def test_update_discount_code
+    discount_code_response = ActiveSupport::JSON.decode(load_fixture('discount_code'))
+    discount_code_response['discount_code']['code'] = "WINTERSALE50"
+    @discount_code.code = "WINTERSALE50"
+    fake 'price_rules/102586120/discount_codes/1002091923', method: :put, status: 200, body: ActiveSupport::JSON.encode(discount_code_response)
```
Comment (felix-d) [inline test/discount_code_test.rb]: missing `@discount_code.save`?

File: test/discount_code_test.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    discount_codes = ShopifyAPI::DiscountCode.all params: { price_rule_id: 102586120 }
+
+    assert_equal 1, discount_codes.length
+    assert_equal 1002091923, discount_codes.first.id
+  end
+
+  def test_create_discount_code
+    fake 'price_rules/102586120/discount_codes', method: :post, status: 201, body: load_fixture('discount_code')
+    discount_code = ShopifyAPI::DiscountCode.new
+    discount_code.prefix_options[:price_rule_id] = 102586120
```
Comment (felix-d) [inline test/discount_code_test.rb]: is that a common pattern? should this implementation detail be abstracted with a setter?

```ruby
module ShopifyAPI
  class DiscountCode < Base
    ...
    def price_rule_id=(id)
      @prefix_options[:price_rule_id] = id
    end
  end
end
...
discount_code.price_rule_id = 102586120
```

Comment (jordanliddle) [inline test/discount_code_test.rb]: It's seems like a common pattern in this lib, but I can certainly look into porting this pattern onto the other resources like [Fulfillment](https://github.com/Shopify/shopify_api/blob/master/lib/shopify_api/resources/fulfillment.rb).

Comment (felix-d) [inline test/discount_code_test.rb]: if it's a common pattern don't worry about this for now

Comment (jordanliddle) [inline test/discount_code_test.rb]: Ok sounds good! Should we 🚢  this when we hit 100% then?

Comment (felix-d) [inline test/discount_code_test.rb]: yep!

**Comment by felix-d** [comment]
shit wrong PR 😄

**Pull request merged**